docs/api-spec.md

```diff
@@ -279,13 +280,16 @@ List responses have the following fields (based on [`meta.v1/ListMeta`](https://
 |--------------|------------|-------------|
 | `name`        | string    | REQUIRED    |
 | `description` | string    | REQUIRED    |
+| `type`      | Enum:<br>- `"string"` (default)<br>- `"array"` <br>- `"object"` | RECOMMENDED (Each of the values is RECOMMENDED.)|
+| `properties`  | map<string,PropertySpec> | RECOMMENDED <br><br>note: `PropertySpec` is a type that defines the spec of an individual key. See how to define the `properties` section in the [example](../examples/v1beta1/taskruns/alpha/object-param-result.yaml).|
 
 ### `TaskRunResult`
 
 | Field Name | Field Type | Requirement |
 |------------|------------|-------------|
 | `name`     | string     | REQUIRED    |
-| `value`    | string     | REQUIRED    |
+| `type`      | Enum:<br>- `"string"` (default)<br>- `"array"` <br>- `"object"` | RECOMMENDED (Each of the values is RECOMMENDED.) |
+| `value`    | `ParamValue`     | REQUIRED    |
```

Comment:
it's a bit unfortunate that we're changing the type of a required field but since we only have one primary implementation of tekton currently it's probably OK?

Reply:
I guess so because it's a backward compatible change. String type result is still supported as the way it was supported before.

Reply:
Just to clarify, if there were other implementations of the Tekton API, this would be a problem. The reason is that they might support the field "value" as a string, but with this change they are "required" to support value as a "ParamValue". This would mean an implementation that was previously conformant would not be conformant after this change. But since we have one main implementation, it doesn't matter too much.

Reply:
Thanks for the clarification!